app/scripts/inpage.js

```diff
@@ -55,9 +58,21 @@ if (shouldInjectProvider()) {
     target: CONTENT_SCRIPT,
   });
 
-  initializeProvider({
+  const MetaMaskEIP6963ProviderInfo = {
+    walletId: 'io.metamask',
+    uuid: uuid(),
+    name: 'MetaMask',
+    icon: 'https://raw.githubusercontent.com/MetaMask/brand-resources/cb6fd847f3a9cc5e231c749383c3898935e62eab/SVG/metamask-fox.svg',
+  };
+
+  const provider = initializeProvider({
     connectionStream: metamaskStream,
     logger: log,
     shouldShimWeb3: true,
   });
+
+  announceProvider({
+    info: MetaMaskEIP6963ProviderInfo,
+    provider,
+  });
```

Comment:
Announce immediately after instantiating the provider object and setting it as `window.ethereum`.